api/v3/VolunteerNeed.php

```diff
@@ -186,3 +186,24 @@ function civicrm_api3_volunteer_need_delete($params) {
   return _civicrm_api3_basic_delete('CRM_Volunteer_BAO_Need', $params);
 }
 
+/**
+ * Set the default getList behavior for Volunteer Need to have the label be the role
+ * and the description be the formatted date and time
+ *
+ * @param array $request
+ *   The parameters passed to the sub-API call (i.e., the parameters to the get
+ *   call underlying the getList call). These are passed to getList in
+ *   $params['params'].
+ * @return array
+ *   Despite the fact that $request represents a subset of the parameters passed
+ *   to getList, the return of this function is merged with the getList params
+ *   in their entirety.
+ */
+function _civicrm_api3_volunteer_need_getlist_defaults(&$request) {
+  return array(
+    'extra' => array('is_flexible', 'role_id', 'role_label', 'start_time', 'display_time'),
```

Comment:
It breaks if you don't include the extra fields. Specifically 'role_label' and 'role_description' display as null unless they and 'role_id' are passed as extra and 'display_time' always displays as 'Any' unless you include start_time and 'is_flexible' makes it so if there is no 'role_label' it displays 'Any' instead of null.